### PR TITLE
Adds --head on HEAD requests

### PR DESCRIPTION
**Pull request by donatj**
The currently generated cURL requests for HEAD do not work.  It throws warnings as follows and never completes.

screen shot 2017-02-20 at 10 57 21 am

This adds --head on HEAD requests to correct the issue:

screen shot 2017-02-20 at 10 57 35 am

### CODE REVIEW

File: curl.mustache [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {{#request.name}}## {{{request.name}}}
 {{/request.name}}{{#request.cURLDescription}}
 {{{request.cURLDescription}}}
-{{/request.cURLDescription}}curl -X "{{{request.method}}}" "{{{request.url}}}" \
+{{/request.cURLDescription}}curl {{#request_is_head}}--head {{/request_is_head}}-X "{{{request.method}}}" "{{{request.url}}}" \
```
Comment (mittsh) [inline curl.mustache]: How about:

```
{{/request.cURLDescription}}curl {{#request_is_head}}--head {{/request_is_head}}{{#specify_method}}-X "{{{request.method}}}" {{/specify_method}}"{{{request.url}}}" \
```

And define:

```js
"specify_method": request.method != "GET" && request.method != "HEAD"
```

So it's not too redundant with `--head` and also we take the chance to add a tiny optimization not specifying `GET` method when it's the default.

**Review by mittsh — COMMENTED** [review]
Thanks a lot. That's a great fix, I didn't think about that edge case on `HEAD` requests. Would you mind considering the suggestion I've made below? I think it will make the generated curl code a bit nicer.

File: curl.mustache (hunk 1 above)
Comment (donatj) [reply curl.mustache]: Not a problem at all, one sec ;)

**Comment by donatj** [comment]
I've made the requested change.

**Review by mittsh — APPROVED** [review]
Beautiful! Thanks a ton!

**Pull request merged**